servlet/src/main/java/io/undertow/servlet/core/ApplicationListeners.java

```diff
@@ -325,7 +326,12 @@ public void sessionDestroyed(final HttpSession session) {
         final HttpSessionEvent sre = new HttpSessionEvent(session);
         for (int i = httpSessionListeners.length - 1; i >= 0; --i) {
             ManagedListener listener = httpSessionListeners[i];
-            this.<HttpSessionListener>get(listener).sessionDestroyed(sre);
+            try {
+                this.<HttpSessionListener> get(listener).sessionDestroyed(sre);
+            } catch (Exception e) {
+                // UNDERTOW-1773 to avoid leak, make sure the iteration continue in case of exception.
+                UndertowLogger.ROOT_LOGGER.debugf(e, "Exception happens in sessionDestroyed for session %s ", session);
```

Comment:
This should really be at error level.

Reply:
Thank you, I have updated the log level to ERROR